advocacy_docs/partner_docs/HashicorpVault/04-ConfiguringHashicorpVault.mdx

```diff
@@ -0,0 +1,142 @@
+
+2. Assume root user.
+
+3. When you are the root user, type ‘vault secrets enable kmip’ in the command line.
+```bash
+root@ip-172-31-46-134:/home/ubuntu# vault secrets enable kmip
+Success! Enabled the kmip secrets engine at: kmip/
+```
+
+4. You will then need to configure the Hashicorp Vault secrets engine with the desired kmip listener address.
+
+5. Enter `vault write kmip/config listen_addrs=0.0.0.0:5696` in the command line.
+```bash
+root@ip-172-31-46-134:/home/ubuntu# vault write kmip/config listen_addrs=0.0.0.0:5696
+Success! Data written to: kmip/config
+```
+
+6. Define the Scopes and Roles to determine what allowed operations a role can perform.
+
+7. Enter `vault write -f kmip/scope/*scope_name*` in the command line.
+```bash
+root@ip-172-31-46-134:/home/ubuntu# vault write -f kmip/scope/edb
+Success! Data written to: kmip/scope/edb
+```
+8. If you ever need to view your scopes you have created you can enter `vault list kmip/scope` in the command line.
+
+9. Define the role for that scope and what operations they are allowed to do.
+
+10. Enter `vault write kmip/scope/*scope_name*/role/*role_name* operation_all=true` in the command line.
+```bash
+root@ip-172-31-46-134:/home/ubuntu# vault write kmip/scope/edb/role/admin operation_all=true
+Success! Data written to: kmip/scope/edb/role/admin
+```
+
+11. You can read your scope and role with this command `vault read kmip/scope/*scope_name*/role/*role_name*`
+```bash
+root@ip-172-31-46-134:/home/ubuntu# vault read kmip/scope/edb/role/admin
+Key                    Value
+---                    -----
+operation_all          true
+tls_client_key_bits    0
+tls_client_key_type    n/a
+tls_client_ttl         0s
+```
+
+## Generate Client Certificates
+
+After a scope and role have been created you will need to generate client certificates that will be used within your pykmip.conf file for key management.  These certificates can be used to establish communication with Hashicorp Vault’s KMIP Server.
+
+1. Generate the client certificate, this will provide the CA Chain, the private key and the certificate.
+
+2. Enter `vault write -f -field=certificate \ kmip/scope/*scope_name*/role/*role_name*/credential/generate > *certificate_name*.pem`.
+```bash
+root@ip-172-31-46-134:/home/ubuntu# vault write -f -field=certificate \ kmip/scope/edb/role/admin/credential/generate > kmip-cert.pem
+```
+
+3. To view your certificates you can type `cat *certificate_name*.pem*` in the command line and this will return the certificates from Hashicorp Vault.
+```bash
+root@ip-172-31-46-134:/home/ubuntu# cat kmip-cert.pem 
+```
+
+4. You will need to separate the individual certificates into .pem files so they can be used in your pykmip.conf file.
+
+5. Create a key.pem file which will contain the private key in the certificate chain.
```

Comment:
could add a note:
Make sure to include 
----BEGIN ------
----END  -------
in the file name

Reply:
This is a good note, I will add it after Step 4 so it can be followed for all the certs that need to be split up.